### PR TITLE
[COASTAL-1395] UI flicker

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/COASTAL-1395

### CODE REVIEW

File: Loop/View Controllers/StatusTableViewController.swift [hunk 1]
```diff
@@ -307,10 +307,6 @@ final class StatusTableViewController: LoopChartsTableViewController {
                 default:
                     break
                 }
-                Task { @MainActor in
-                    refreshContext.update(with: .status)
-                    await reloadData(animated: true)
-                }
```
Comment (nhamming) [inline Loop/View Controllers/StatusTableViewController.swift:313]: This is the solution for the flicker our partners reported. Note that bolus progress is being updated with `updateBolusProgress`.

File: Loop/View Controllers/StatusTableViewController.swift [hunk 2]
```diff
@@ -2120,14 +2113,8 @@ extension StatusTableViewController: CompletionDelegate {
...
-        Task { @MainActor in
-
-            basalDeliveryState = status.basalDeliveryState
-            bolusState = status.bolusState
-
-            refreshContext.update(with: .status)
-            await self.reloadData(animated: true)
-        }
+        basalDeliveryState = status.basalDeliveryState
+        bolusState = status.bolusState
```
Comment (nhamming) [inline Loop/View Controllers/StatusTableViewController.swift:2117]: This was calling `reloadData()` twice. First was during the update of the `bolusState` or `basalDeliveryState`.

File: Loop/View Controllers/StatusTableViewController.swift [hunk 3]
```diff
@@ -611,7 +607,6 @@ final class StatusTableViewController: LoopChartsTableViewController {
             self.currentCOBDescription = nil
         }
 
-        self.tableView.beginUpdates()
```
Comment (nhamming) [inline Loop/View Controllers/StatusTableViewController.swift:614]: this is an unneeded duplication.

File: Loop/View Controllers/StatusTableViewController.swift [hunk 4]
```diff
@@ -641,8 +636,6 @@ final class StatusTableViewController: LoopChartsTableViewController {
 
         redrawCharts()
 
-        tableView.endUpdates()
-
```
Comment (nhamming) [inline Loop/View Controllers/StatusTableViewController.swift:645]: this is an unneeded duplication

File: Loop/Managers/SettingsManager.swift [hunk 5]
```diff
@@ -177,6 +177,12 @@ class SettingsManager {
     ///
     /// - Parameter timeZone: The time zone
     func setScheduleTimeZone(_ timeZone: TimeZone) {
+        let shouldUpdate = settings.basalRateSchedule?.timeZone != timeZone ||
+        settings.carbRatioSchedule?.timeZone != timeZone ||
+        settings.insulinSensitivitySchedule?.timeZone != timeZone ||
+        settings.glucoseTargetRangeSchedule?.timeZone != timeZone
+        guard shouldUpdate else { return }
+
```
Comment (nhamming) [inline Loop/Managers/SettingsManager.swift:185]: This is to reduce unneeded calls of ` .LoopDataUpdated`.

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**